### ISSUE TITLE
Skip parameter not working properly

### ISSUE THREAD

**Issue by Titozzz** [apollographql/apollo-client#9765]
Hello @benjamn, here is another one for you 🤞🏻 

**Intended outcome:**

Setting `skip: true` on a query should prevent its execution no matter what. Sometime variables are not available yet and we do not want to fetch the back-end without them to avoid errors.

**Actual outcome:**

There is a bug in a specific configuration that makes the query to the back-end, but doesn't get the result back with `skip:true`

**How to reproduce the issue:**

Here is a link to a [codesandbox](https://codesandbox.io/s/apollo-client-3-5-9-bug-forked-5zbl63?file=/src/App.js).
TLDR:
- define custom fetch policies
- be on version >= 3.6.3
- have `skip: false`, but update the `variables` -> an invisible network request is made (invisible cause even if it succeeds you wont get the results)

Where the bug probably comes from: [this](https://github.com/apollographql/apollo-client/pull/9665) new skip management.

**Versions**

Apollo Client 3.6.6 (bug appeared on 3.6.3, before that skip was even more bugged)

**Comment by denisk20** [comment]
To me `skip: true` doesn't work with combination of `fetchPolicy: 'cache-and-network'` and `nextFetchPolicy: 'cache-only'`. Here's a branch which shows that.

https://github.com/denisk20/apolloTest/tree/skipIssue

On the first click of a Button (current page = 2) a network request can be seen in Flipper, even though `skip: page < 5`.
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/3080318/171904416-ff5f5e39-7223-471a-9567-c73ccd03865c.png">

**Comment by denisk20** [comment]
To me `skip: true` doesn't work with combination of `fetchPolicy: 'cache-and-network'` and `nextFetchPolicy: 'cache-only'`. Here's a branch which shows that.

https://github.com/denisk20/apolloTest/tree/skipIssue

On the first click of a Button (current page = 2) a network request can be seen in Flipper, even though `skip: page < 5`.
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/3080318/171904416-ff5f5e39-7223-471a-9567-c73ccd03865c.png">

**Comment by hwillson** [comment]
@Titozzz I tried your reproduction with `@apollo/client@3.6.7` and only see a network request happen when the `offset` hits 2 (as expected). Can you confirm that everything looks okay when you try?

**Comment by Titozzz** [comment]
I'll try in an hour

**Comment by Titozzz** [comment]
@hwillson I'm very sorry, the sandbox didn't include the nextFetchPolicy. Here is one showing the issue (still reproduces with 3.6.8).
https://codesandbox.io/s/apollo-client-3-6-3-skip-bug-5zbl63

**Comment by benjamn** [comment]
@Titozzz Fix incoming! #9823

**Comment by benjamn** [comment]
@Titozzz (and others) Please try running `npm i @apollo/client@next` to get version 3.6.9 (published to npm as `next`, but not yet `latest`). I can confirm the reproduction from https://github.com/apollographql/apollo-client/issues/9765#issuecomment-1153882328 works now (no extra network requests), but it would be great to make sure this release doesn't introduce any new problems before we promote it to `latest`! 🙌

**Comment by Titozzz** [comment]
@benjamn All good on my side during local testing. Will come back to you if we encounter issues in production
